hexrd/ui/main_window.py

```diff
@@ -563,24 +563,33 @@ def on_action_edit_euler_angle_convention(self):
         self.update_all()
         self.update_config_gui()
 
-    def on_action_edit_apply_polar_mask_triggered(self):
+    def on_action_edit_apply_hand_drawn_mask_triggered(self):
         # Make the dialog
         canvas = self.ui.image_tab_widget.image_canvases[0]
-        self._apply_polar_mask_line_picker = (
+        self._apply_drawn_mask_line_picker = (
             HandDrawnMaskDialog(canvas, self.ui))
-        self._apply_polar_mask_line_picker.start()
-        self._apply_polar_mask_line_picker.finished.connect(
+        self._apply_drawn_mask_line_picker.start()
+        self._apply_drawn_mask_line_picker.finished.connect(
             self.run_apply_polar_mask)
 
-    def run_apply_polar_mask(self, line_data):
+    def run_apply_polar_mask(self, dets, line_data):
         raw_lines = HexrdConfig().raw_masks_line_data
         polar_lines = HexrdConfig().polar_masks_line_data
-        for line in line_data:
-            name = unique_name({**raw_lines, **polar_lines}, 'polar_mask_0')
-            HexrdConfig().polar_masks_line_data[name] = [line.copy()]
-            HexrdConfig().visible_masks.append(name)
-            create_polar_mask([line.copy()], name)
-        HexrdConfig().polar_masks_changed.emit()
+        if self.image_mode == ViewType.raw:
+            for det, line in zip(dets, line_data):
+                name = unique_name({**raw_lines, **polar_lines}, 'raw_mask_0')
+                HexrdConfig().raw_masks_line_data[name] = [(det, line.copy())]
```

Comment:
I guess we need to update some of these variables that changed via #1044?

Reply:
Yes, I've merged master in and cleaned up the resulting conflicts.